### PR TITLE
feature added, move-column-left-or-monitor-left and move-column-right…

### PR DESCRIPTION
**Pull request by exoess**
feature added, move-column-left-or-monitor-left and move-column-right-or-monitor-right

based on #456

### CODE REVIEW

File: src/input/mod.rs [hunk 1]
```diff
@@ -571,6 +571,40 @@ impl State {
...
+                } else {
+                    self.niri.layout.move_left();
+                    self.maybe_warp_cursor_to_focus();
+                }
+
+                // FIXME: granular
+                self.niri.queue_redraw_all();
+            }
+            Action::MoveColumnRightOrMonitorRight => {
+                if let Some(output) = self.niri.output_left() {
```
Comment (Suyashtnt) [inline src/input/mod.rs:592]: wouldn't this need to be `self.niri.output_right()`?

Comment (exoess) [reply src/input/mod.rs]: fixed

**Comment by SalmanFarooqShiekh** [comment]
`move-column-left-or-to-monitor-{left,right}`?

File: src/input/mod.rs (hunk 1 above)
Comment (SalmanFarooqShiekh) [reply src/input/mod.rs]: You might want to mark this as resolved.

**Comment by SalmanFarooqShiekh** [comment]
Will you also do `move-window-{up,down}-or-to-monitor-{up,down}` similar to https://github.com/YaLTeR/niri/pull/497/files?

**Comment by exoess** [comment]
prob not, it's not personally useful to me, and i don't even know if i implemented this right

File: niri-config/src/lib.rs [hunk 2]
```diff
@@ -977,6 +977,8 @@ pub enum Action {
     MoveColumnRight,
     MoveColumnToFirst,
     MoveColumnToLast,
+    MoveColumnLeftOrMonitorLeft,
+    MoveColumnRightOrMonitorRight,
```
Comment (YaLTeR) [inline niri-config/src/lib.rs:981]:
```suggestion
    MoveColumnLeftOrToMonitorLeft,
    MoveColumnRightOrToMonitorRight,
```
For grammar and consistency sake (like i.e. `MoveWindowDownOrToWorkspaceDown`)

File: niri-ipc/src/lib.rs [hunk 3]
```diff
@@ -152,6 +152,10 @@ pub enum Action {
     MoveColumnToFirst,
     /// Move the focused column to the end of the workspace.
     MoveColumnToLast,
+    /// Move the focused column to the left or to the left monitor
+    MoveColumnLeftOrMonitorLeft,
+    /// Move the focused column to the right or to the right monitor
+    MoveColumnRightOrMonitorRight,
```
Comment (YaLTeR) [inline niri-ipc/src/lib.rs:158]:
```suggestion
    /// Move the focused column to the left or to the monitor to the left.
    MoveColumnLeftOrToMonitorLeft,
    /// Move the focused column to the right or to the monitor to the right.
    MoveColumnRightOrToMonitorRight,
```

File: src/layout/mod.rs [hunk 4]
```diff
@@ -1196,6 +1196,36 @@ impl<W: LayoutElement> Layout<W> {
         monitor.move_column_to_last();
     }
 
+    pub fn move_column_left_or_output(&mut self, output: &OutputId) -> bool {
```
Comment (YaLTeR) [inline src/layout/mod.rs:1199]:
```suggestion
    pub fn move_column_left_or_to_output(&mut self, output: &Output) -> bool {
```
These should accept `&Output` not `&OutputId`.

File: src/layout/mod.rs [hunk 5]
```diff
@@ -1196,6 +1196,36 @@ impl<W: LayoutElement> Layout<W> {
...
+        self.move_column_to_output(output);
+        true
+    }
+
+    pub fn move_column_right_or_output(&mut self, output: &OutputId) -> bool {
+        if let Some(monitor) = self.active_monitor() {
+            let workspace = monitor.active_workspace();
+            let curr_idx = workspace.active_column_idx;
+
+            if !workspace.columns.is_empty() && curr_idx != 0 {
```
Comment (YaLTeR) [inline src/layout/mod.rs:1219]:
```suggestion
            if !workspace.columns.is_empty() && curr_idx != workspace.columns.len() - 1 {
```

File: src/layout/mod.rs [hunk 6]
```diff
@@ -2797,6 +2827,8 @@ mod tests {
         MoveColumnRight,
         MoveColumnToFirst,
         MoveColumnToLast,
+        MoveColumnLeftOrMonitorLeft(#[proptest(strategy = "1..=2u8")] u8),
+        MoveColumnRightOrMonitorRight(#[proptest(strategy = "1..=2u8")] u8),
```
Comment (YaLTeR) [inline src/layout/mod.rs:2831]:
```suggestion
        MoveColumnLeftOrToMonitorLeft(#[proptest(strategy = "1..=2u8")] u8),
        MoveColumnRightOrToMonitorRight(#[proptest(strategy = "1..=2u8")] u8),
```

**Review by YaLTeR — CHANGES_REQUESTED** [review]
Almost good to go, just some grammatical renames and a condition fix.

**Comment by exoess** [comment]
@YaLTeR, added your fixes, should be ready now

**Comment by YaLTeR** [comment]
Looks like you've changed the lines that I commented about but not the other lines that depend on them, or the same name used elsewhere

**Comment by exoess** [comment]
oh yea, sorry

File: niri-config/src/lib.rs [hunk 7]
```diff
@@ -1057,6 +1059,10 @@ impl From<niri_ipc::Action> for Action {
             niri_ipc::Action::MoveColumnRight => Self::MoveColumnRight,
             niri_ipc::Action::MoveColumnToFirst => Self::MoveColumnToFirst,
             niri_ipc::Action::MoveColumnToLast => Self::MoveColumnToLast,
+            niri_ipc::Action::MoveColumnLeftOrToMonitorLeft => Self::MoveColumnLeftOrToMonitorLeft,
+            niri_ipc::Action::MoveColumnRightOrToMonitorRight => {
```
Comment (SalmanFarooqShiekh) [inline niri-config/src/lib.rs:1063]: niri_ipc::Action::MoveColumnRightOrToMonitorRight => Self::MoveColumnRightOrToMonitorRight,

Comment (exoess) [reply niri-config/src/lib.rs:1063]: this does not follow rustfmt

**Review by YaLTeR — APPROVED** [review]
Looks good.

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**